common/src/main/kotlin/entity/Interactions.kt

```diff
@@ -395,3 +399,51 @@ sealed class InteractionResponseType(val type: Int) {
 
     }
 }
+
+
+@KordPreview
+@Serializable
+data class DiscordGuildApplicationCommandPermissions(
+    val id: Snowflake,
+    @SerialName("application_id")
+    val applicationId: Snowflake,
+    @SerialName("guild_id")
+    val guildId: Snowflake,
+    val permissions: List<DiscordGuildApplicationCommandPermission>
+)
+
+@KordPreview
+@Serializable
+data class PartialDiscordGuildApplicationCommandPermissions(
+    val id: Snowflake,
+    val permissions: List<DiscordGuildApplicationCommandPermission>
+)
+
+@KordPreview
+@Serializable
+data class DiscordGuildApplicationCommandPermission(
+    val id: Snowflake,
+    val type: Type,
+    val permission: Boolean
+) {
+    @Serializable(with = Type.Serializer::class)
+    sealed class Type(val value: Int) {
+        object Role : Type(1)
+        object User : Type(2)
+        object Unknown : Type(-1)
```

Comment:
Unknown should be a class that takes the unknown number, where you pass the decoded int if it didn't match any of the known types

Reply:
Change this to Something more expresive than `Type` and see if you can create a test for permissions

Reply:
imo `DiscordGuildApplicationCommandPermission.Type` is very expressive

Reply:
PermissionType might be a better name

Reply:
I guess that's not a problem either way